ocaml/xapi/cli_operations.ml

```diff
@@ -4340,12 +4340,23 @@ let update_upload fd printer rpc session_id params =
     let sr =
       if List.mem_assoc "sr-uuid" params
       then Client.SR.get_by_uuid rpc session_id (List.assoc "sr-uuid" params)
-      else Client.Pool.get_default_SR ~rpc ~session_id ~self:(List.hd pools)
-  in
+      else begin
+        let sr = Client.Pool.get_default_SR ~rpc ~session_id ~self:(List.hd pools) in
+        let ref_is_valid = Server_helpers.exec_with_new_task
+            ~session_id "Checking default SR validity"
+            (fun __context -> Db.is_valid_ref __context sr) in
+        if not ref_is_valid then
```

Comment:
A small tip regarding code style: I think the logic ```if ref_is_valid then sr else failwith "..."``` is generally more legible. But you don't need to change it.